libcnb-data/src/launch.rs

```diff
@@ -65,30 +66,102 @@ pub struct Label {
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub direct: Option<bool>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub default: Option<bool>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub args: Vec<String>,
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    pub direct: bool,
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    pub default: bool,
+}
+
+pub struct ProcessBuilder {
+    process: Process,
 }
 
-impl Process {
-    pub fn new(
-        r#type: ProcessType,
-        command: impl Into<String>,
-        args: Option<impl IntoIterator<Item = impl Into<String>>>,
-        direct: Option<bool>,
-        default: Option<bool>,
-    ) -> Self {
+/// A non-consuming builder for [`Process`] values.
+///
+/// # Examples
+/// ```
+/// # use libcnb_data::process_type;
+/// # use libcnb_data::launch::ProcessBuilder;
+/// ProcessBuilder::new(process_type!("web"), "java")
+///     .arg("-jar")
+///     .arg("target/application-1.0.0.jar")
+///     .default(true)
+///     .build();
+/// ```
+impl ProcessBuilder {
+    /// Constructs a new `ProcessBuilder` with the following defaults:
+    ///
+    /// * No arguments to the process
+    /// * `direct` is `false`
+    /// * `default` is `false`
+    pub fn new(r#type: ProcessType, command: impl Into<String>) -> Self {
         Self {
-            r#type,
-            command: command.into(),
-            args: args.map(|args| args.into_iter().map(std::convert::Into::into).collect()),
-            direct,
-            default,
+            process: Process {
+                r#type,
+                command: command.into(),
+                args: Vec::new(),
+                direct: false,
+                default: false,
+            },
         }
     }
+
+    /// Adds an argument to the process.
+    ///
+    /// Only one argument can be passed per use. So instead of:
+    /// ```no_run
+    /// # use libcnb_data::process_type;
+    /// # libcnb_data::launch::ProcessBuilder::new(process_type!("web"), "command")
+    /// .arg("-C /path/to/repo")
+    /// # ;
+    /// ```
+    ///
+    /// usage would be:
+    ///
+    /// ```no_run
+    /// # use libcnb_data::process_type;
+    /// # libcnb_data::launch::ProcessBuilder::new(process_type!("web"), "command")
+    /// .arg("-C")
+    /// .arg("/path/to/repo")
+    /// # ;
+    /// ```
+    ///
+    /// To pass multiple arguments see [`args`](Self::args).
+    pub fn arg(&mut self, arg: impl Into<String>) -> &mut Self {
+        self.process.args.push(arg.into());
+        self
+    }
+
+    /// Adds multiple arguments to pass to the process.
+    ///
+    /// To pass a single argument see [`arg`](Self::arg).
+    pub fn args(&mut self, args: impl IntoIterator<Item = impl Into<String>>) -> &mut Self {
+        for arg in args {
+            self.arg(arg);
+        }
+
+        self
+    }
+
+    /// Sets the direct flag on the process.
```

Comment:
(just since I think it may be non-obvious to people who haven't read the spec, what the implications of this bool are)

```suggestion
    /// Sets the `direct` flag on the process.
    ///
    /// If this is true, the lifecycle will launch the command directly, rather than via a shell.
```